### ISSUE TITLE
Shell parameter expansion in tag precludes linting

### ISSUE THREAD

**Issue by zaventh** [hadolint/hadolint#965]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Hadolint should support linting a Dockerfile with a shell parameter expansion in a `FROM` directive.

### Actual behavior

Hadolint fails with message:

```shell
$ hadolint Dockerfile 
Dockerfile:3:28 unexpected ':' expecting '@', '\', a new line followed by the next instruction, or the image tag
```

### Steps to reproduce the behavior

Install hadolint:

```shell
$ hadolint --version
Haskell Dockerfile Linter 2.12.0
```

Create a `Dockerfile` that begins as follows:

```dockerfile
ARG BUILD_VER
FROM alpine/k8s:${BUILD_VER:-1.24.10}
```

> Note the actual image referenced is irrelevant.

Docker build succeeds and supports this kind of use case, however hadolint fails and there does not appear to be a way to disable this kind of check. The only remedies are to remove the parameter expansion directive entirely or disable linting of the affected `Dockerfile`.

This is similar to, but not exactly the same as, other issues such as https://github.com/hadolint/hadolint/issues/355. In this case the proposed PR there will not remedy this issue.
